packages/iov-ethereum/src/integrationtests/atomicswap.spec.ts

```diff
@@ -0,0 +1,324 @@
+    expect(offer.kind).toEqual(SwapProcessState.Open);
+    expect(offer.data.recipient).toEqual(this.receiveAddress);
+    expect(offer.data.amounts.length).toEqual(1);
+    expect(offer.data.amounts[0]).toEqual({
+      quantity: "2000000000000000000",
+      fractionalDigits: 18,
+      tokenTicker: ETH,
+    });
+
+    // send counteroffer
+    const counter = await this.connection.withDefaultFee<SwapOfferTransaction>({
+      kind: "bcp/swap_offer",
+      creator: this.senderIdentity,
+      recipient: recipient,
+      amounts: [amount],
+      swapId: await AtomicSwapHelpers.createId(),
+      hash: offer.data.hash,
+      timeout: {
+        height: (await this.connection.height()) + 5,
+      },
+    });
+    const nonce = await this.connection.getNonce({ pubkey: this.senderIdentity.pubkey });
+    const signed = await this.profile.signTransaction(counter, ethereumCodec, nonce);
+    const postable = await ethereumCodec.bytesToPost(signed);
+    const post = await this.connection.postTx(postable);
+    const blockInfo = await post.blockInfo.waitFor(info => !isBlockInfoPending(info));
+    if (!isBlockInfoSucceeded(blockInfo)) {
+      throw new Error("Transaction failed");
+    }
+  }
+
+  public async claimFromKnownPreimage(): Promise<void> {
+    // review counteroffer
+    const swaps = await this.connection.getSwaps({ recipient: this.receiveAddress });
+    const counter = swaps[swaps.length - 1];
+    expect(counter.kind).toEqual(SwapProcessState.Open);
+    expect(counter.data.recipient).toEqual(this.receiveAddress);
+    expect(counter.data.amounts.length).toEqual(1);
+    expect(counter.data.amounts[0]).toEqual({
+      quantity: "5000000000000000000",
+      fractionalDigits: 18,
+      tokenTicker: ETH,
+    });
+
+    // claim funds
+    const claim = await this.connection.withDefaultFee<SwapClaimTransaction>({
+      kind: "bcp/swap_claim",
+      creator: this.receiverIdentity,
+      swapId: counter.data.id,
+      preimage: this.preimage!,
+    });
+    const nonce = await this.connection.getNonce({ pubkey: this.receiverIdentity.pubkey });
+    const signed = await this.profile.signTransaction(claim, ethereumCodec, nonce);
+    const postable = await ethereumCodec.bytesToPost(signed);
+    const post = await this.connection.postTx(postable);
+    const blockInfo = await post.blockInfo.waitFor(info => !isBlockInfoPending(info));
+    if (!isBlockInfoSucceeded(blockInfo)) {
+      throw new Error("Transaction failed");
+    }
+  }
+
+  public async claimFromRevealedPreimage(): Promise<void> {
+    // find claim
+    const claimSwaps = await this.connection.getSwaps({ sender: this.sendAddress });
+    const claim1 = claimSwaps[claimSwaps.length - 1];
+    expect(claim1.kind).toEqual(SwapProcessState.Claimed);
+
+    const offerSwaps = await this.connection.getSwaps({ recipient: this.receiveAddress });
+    const offer = offerSwaps[offerSwaps.length - 1];
+    expect(offer.kind).toEqual(SwapProcessState.Open);
```

Comment:
I'm not super happy about the expect() calls inside of the `Actor` class. I know this works and why and – shame on me – I also did this in the other file. But I think this belongs into the `it` body directly or should be converted into an exception.

Reply:
I definitely agree with this, I was just following the pattern of `iov-core`.